### PR TITLE
Feature/aoeff interpreter migration

### PR DESCRIPTION
**Pull request by rodrigolimoes**
# Pull Request

## Description
Migrated "AoeffInterpreter" file to TypeScript, added unit tests and added JSDOC documentation.

## Test
1. Navigate to the project folder
  ~~~console
  cd <rootfolder>/archimate-model-importer
  ~~~
2. Run the test script
~~~console
npm run test
// or
npm run test:unique 'AoeffInterpreter'
~~~

### CODE REVIEW

File: __tests__/unit/lib/models/Model.test.ts [hunk 1]
```diff
@@ -33,13 +33,13 @@ describe('Model', () => {
       identifier: 'bc8c928b-dafb-4e61-91b3-7c3e5b93a900',
       sourceId: 'd0c22546-6ae6-4ba9-a141-222cc6eea16d',
       targetId: 'cc07d17e-8450-4adf-84d1-ea7d92ec01ab',
-      type: RelationshipType.Aggregation,
+      type: RelationshipType.Aggregation.toLowerCase(),
```
Comment (Diorbert) [inline __tests__/unit/lib/models/Model.test.ts:36]: Hi, @rodrigolimoes . Can we change the enum values to be in lowercase?

File: src/lib/common/enums/relationshipType.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-  Triggering = 'triggering',
-  Access = 'access',
-  Influence = 'influence',
-  Realization = 'realization',
-  Realisation = 'realisation',
-  Specialization = 'specialization',
-  Flow = 'flow',
-  Composition = 'composition',
-  Aggregation = 'aggregation',
+  Association = 'Association',
```
Comment (Diorbert) [inline src/lib/common/enums/relationshipType.ts:2]: Is there a particular reason to not keep the values in lowercase?

**Review by Diorbert — APPROVED** [review]
Hi, @rodrigolimoes . Approved! Great work.

I left some comments, though. Let's take a look later.

File: __tests__/unit/lib/models/Model.test.ts (hunk 1 above)
Comment (rodrigolimoes) [reply __tests__/unit/lib/models/Model.test.ts:36]: Hi, @Diorbert.
Yes, we can change the enum values to be lowercase.
I am changing this in another branch.

File: src/lib/common/enums/relationshipType.ts (hunk 2 above)
Comment (rodrigolimoes) [reply src/lib/common/enums/relationshipType.ts:2]: Hi, @Diorbert.
I made a mistake changing these values to uppercase and fixed it in the branch I'm currently working on. These values refer to the translated relationship type in the model and must be kept in lower case.

**Pull request merged**